packages/scheduler/lib/models/tasks.ts

```diff
@@ -305,3 +306,36 @@ export async function expiresIfTimeout(db: knex.Knex): Promise<Result<Task[]>> {
         return Err(new Error(`Error expiring tasks: ${stringifyError(err)}`));
     }
 }
+
+export async function hardDeleteOlderThanNDays(db: knex.Knex, days: number): Promise<Result<Task[]>> {
+    try {
+        // Delete terminated tasks where lastStateTransitionAt is older than N days
+        // unless it is the most recent task for an given schedule
+        const tasks = await db
+            .from<DbTask>(TASKS_TABLE)
+            .where(
+                'id',
+                '=',
+                // schedules.id is null if the task is not associated with a schedule (ie: actions)
+                // or if the task is not the last task for its associated schedule
+                db.raw(`ANY(ARRAY(
+                    SELECT ${TASKS_TABLE}.id
+                    FROM ${TASKS_TABLE}
+                    LEFT JOIN ${SCHEDULES_TABLE} ON ${SCHEDULES_TABLE}.last_scheduled_task_id = ${TASKS_TABLE}.id
+                    WHERE ${TASKS_TABLE}.terminated = true
+                    AND ${TASKS_TABLE}.starts_after < NOW() - INTERVAL '${days} days'
+                    AND ${SCHEDULES_TABLE}.id IS NULL
+                    ORDER BY ${TASKS_TABLE}.id ASC
+                    LIMIT 1000
```

Comment:
1000 is arbitrary. We peak at 600+ new tasks per minutes today for instance. 
This query will running every 10s and delete 1000 entries which means deleting ~6000 entries a minutes so we have some margin